lib/ModelMixins/TableMixin.ts

```diff
@@ -641,63 +670,56 @@ function TableMixin<T extends Constructor<Model<TableTraits>>>(Base: T) {
+          styleFunc: function(feature: any) {
+            const featureRegion = feature.properties[regionType.regionProp];
+            const regionIdString =
+              featureRegion !== undefined && featureRegion !== null
+                ? featureRegion.toString()
+                : "";
+
+            let rowNumber = catalogItem.getImageryLayerFilteredRows(
+              input,
+              currentTimeRows,
+              valuesAsRegions.regionIdToRowNumbersMap.get(
+                regionIdString.toLowerCase()
+              )
+            );
+            let value: string | number | null = isDefined(rowNumber)
+              ? valueFunction(rowNumber)
+              : null;
 
-              return {
-                fillStyle: color.toCssColorString(),
-                strokeStyle: baseMapContrastColor,
-                lineWidth: 1,
-                lineJoin: "miter"
-              };
-            },
-            subdomains: regionType.serverSubdomains,
-            rectangle:
-              Array.isArray(regionType.bbox) && regionType.bbox.length >= 4
-                ? Rectangle.fromDegrees(
-                    regionType.bbox[0],
-                    regionType.bbox[1],
-                    regionType.bbox[2],
-                    regionType.bbox[3]
-                  )
-                : undefined,
-            minimumZoom: regionType.serverMinZoom,
-            maximumNativeZoom: regionType.serverMaxNativeZoom,
-            maximumZoom: regionType.serverMaxZoom,
-            uniqueIdProp: regionType.uniqueIdProp,
-            featureInfoFunc: (feature: any) =>
-              this.getImageryLayerFeatureInfo(input, feature, currentTimeRows)
-          }),
-          show: this.show,
-          clippingRectangle: this.clipToRectangle
-            ? this.cesiumRectangle
-            : undefined
-        };
+            const color = colorMap.mapValueToColor(value);
+            if (color === undefined) {
+              return undefined;
+            }
+
+            return {
+              fillStyle: color.toCssColorString(),
+              strokeStyle: baseMapContrastColor,
+              lineWidth: 1,
+              lineJoin: "miter"
+            };
+          },
+          subdomains: regionType.serverSubdomains,
+          rectangle:
+            Array.isArray(regionType.bbox) && regionType.bbox.length >= 4
+              ? Rectangle.fromDegrees(
+                  regionType.bbox[0],
+                  regionType.bbox[1],
+                  regionType.bbox[2],
+                  regionType.bbox[3]
+                )
+              : undefined,
+          minimumZoom: regionType.serverMinZoom,
+          maximumNativeZoom: regionType.serverMaxNativeZoom,
+          maximumZoom: regionType.serverMaxZoom,
+          uniqueIdProp: regionType.uniqueIdProp,
+          featureInfoFunc: (feature: any) =>
+            this.getImageryLayerFeatureInfo(input, feature, currentTimeRows)
+        });
```

Comment:
All I've done here is split the `ImageryProvider` from the `ImageryParts`. (see `this.regionMappedImageryParts` and `this.regionMappedImageryProvider`)

See this is done so that we aren't re-creating the imageryProvider if things like `opacity` and `show` change